core/trino-main/src/main/java/io/trino/connector/system/TableCommentSystemTable.java

```diff
@@ -159,13 +157,27 @@ private RelationComment getRelationComment(Session session, QualifiedObjectName
             return new RelationComment(true, view.get().getComment());
         }
 
-        RedirectionAwareTableHandle redirectionAware = metadata.getRedirectionAwareTableHandle(session, relationName);
-        if (redirectionAware.tableHandle().isPresent()) {
-            // TODO (https://github.com/trinodb/trino/issues/18514) this should consult accessControl on redirected name. Leaving for now as-is.
-            return new RelationComment(true, metadata.getTableMetadata(session, redirectionAware.tableHandle().get()).getMetadata().getComment());
+        return getCommentOnRedirectedTable(session, relationName);
```

Comment:
"getTableCommentRedirectionAware" or "getTableComment"

(at this point we don't know whether table is redirected)